lib/phoenix_live_view/logger.ex

```diff
@@ -0,0 +1,196 @@
+defmodule Phoenix.LiveView.Logger do
+  @moduledoc """
+  Instrumenter to handle logging of `Phoenix.LiveView` and `Phoenix.LiveComponent` life-cycle events.
+
+  ## Installation
+
+  Add the following line to your `application.ex`:
+
+  ```elixir
+  # lib/my_app/application.ex
+
+  @impl true
+  def start(_type, _args) do
+    # ...
+    Phoenix.LiveView.Logger.install()
+    # ...
+  end
+  ```
+
+  ## Configuration
+
+  The log level is configurable for each `Phoenix.LiveView` module:
+
+  ```elixir
+  use Phoenix.LiveView, log: :debug
+  ```
+
+  To disable logging entirely:
+
+  ```elixir
+  use Phoenix.LiveView, log: false
+  ```
+
+  By default, all life-cycle events are logged as `:info`.
+
+  ## Telemetry
+
+  The following `Phoenix.LiveView` and `Phoenix.LiveComponent` events are logged:
+
+  - `[:phoenix, :live_view, :mount, :stop]`
+  - `[:phoenix, :live_view, :handle_params, :stop]`
+  - `[:phoenix, :live_view, :handle_event, :stop]`
+  - `[:phoenix, :live_view, :handle_info, :stop]`
+  - `[:phoenix, :live_component, :handle_event, :stop]`
+
+  See the [Telemetry](./guides/server/telemetry.md) guide for more information.
+
+  ## Parameter filtering
+
+  If enabled, `Phoenix.LiveView.Logger` will filter parameters based on the configuration of `Phoenix.Logger`. 
+  """
+
+  import Phoenix.Logger, only: [duration: 1, filter_values: 1]
+
+  require Logger
+
+  @doc false
+  def install do
+    handlers = %{
+      [:phoenix, :live_view, :mount, :stop] => &live_view_mount_stop/4,
```

Comment:
We want `&__MODULE__.live_view_mount_stop/4` for these otherwise telemetry will warn about performance for captures

Reply:
TIL, thanks for the explanation 🙂